src/inc/til/at.h

```diff
@@ -5,17 +5,47 @@
 
 namespace til
 {
+    namespace details
+    {
+        // This was lifted from gsl::details::is_span.
+        template<class T>
+        struct is_span_oracle : std::false_type
+        {
+        };
+
+        template<class ElementType, std::size_t Extent>
+        struct is_span_oracle<gsl::span<ElementType, Extent>> : std::true_type
+        {
+        };
+
+        template<class T>
+        struct is_span : public is_span_oracle<std::remove_cv_t<T>>

```

Comment:
Alright well I'll file this PR under the list of "C++ is a made-up witchcraft language"